### ISSUE TITLE
symbol not found when emulating a binary from firmware

### ISSUE THREAD

**Issue by cq674350529** [qilingframework/qiling#762]
**Describe the bug**
When emulating a binary from firmware, the following error occurred.

```
...
[+]	bytearray(b': ')
[+]	bytearray(b'/lib/libgcc_s.so.1')
[+]	bytearray(b': ')
[+]	bytearray(b"version `GLIBC_2.8\' not found")
[+]	bytearray(b'\n')
[+]	writev() = 0x77
[+]	[+] Received Interupt: 2 Hooked Interupt: 2
[+]	0x047c907c: writev(fd = 0x2, vec = 0x7ff3c7f0, vien = 0x6)
[+]	writev() CONTENT:
[+]	...
[+]	bytearray(b': ')
[+]	bytearray(b'/lib/libgcc_s.so.1')
[+]	bytearray(b': ')
[+]	bytearray(b"version `GLIBC_2.17\' not found")
[+]	bytearray(b'\n')
[+]	writev() = 0x78
[+]	[+] Received Interupt: 2 Hooked Interupt: 2
[+]	0x047c907c: writev(fd = 0x2, vec = 0x7ff3c7f0, vien = 0x6)
[+]	writev() CONTENT:
[+]	...
[+]	bytearray(b': ')
[+]	bytearray(b'/lib/libgcc_s.so.1')
[+]	bytearray(b': ')
[+]	bytearray(b"version `GLIBC_2.4\' not found")
[+]	bytearray(b'\n')
[+]	writev() = 0x77
[+]	[+] Received Interupt: 2 Hooked Interupt: 2
[+]	0x047c907c: writev(fd = 0x2, vec = 0x7ff3c7f0, vien = 0x6)
[+]	writev() CONTENT:
[+]	...
[+]	bytearray(b': ')
[+]	bytearray(b'/lib/libgcc_s.so.1')
[+]	bytearray(b': ')
[+]	bytearray(b"version `GLIBC_2.4\' not found")
[+]	bytearray(b'\n')
[+]	writev() = 0x77
[+]	[+] Received Interupt: 2 Hooked Interupt: 2
[+]	0x047c907c: writev(fd = 0x2, vec = 0x7ff3c7f8, vien = 0x6)
[+]	writev() CONTENT:
[+]	...
[+]	bytearray(b': ')
[+]	bytearray(b'/lib/libgcc_s.so.1')
[+]	bytearray(b': ')
[+]	bytearray(b"version `GLIBC_2.4\' not found (required by /lib/libgcc_s.so.1)")
...
```

**Sample Code**
```python
 rootfs_path = "./test_rootfs"
 bin_path = os.path.join(rootfs_path, "lldpd")
 ql = Qiling([bin_path, "-h"], rootfs_path, console=True, verbose=QL_VERBOSE.DEBUG)

 ql.run()
```

The related rootfs is attached! Thanks.
[test_rootfs.zip](https://github.com/qilingframework/qiling/files/6277612/test_rootfs.zip)

**Comment by cq674350529** [comment]
Another rootfs is attached, which seems to have the same issue.
[test_rootfs_rv16x.zip](https://github.com/qilingframework/qiling/files/6323840/test_rootfs_rv16x.zip)

**Comment by zi0Black** [comment]
I'm experiencing the same issue with two different ARM binary and associated rootfs.

**Comment by wtdcode** [comment]
> I'm experiencing the same issue with two different ARM binary and associated rootfs.

Could you post your binary here? I'm stuck with the two rootfs above.

**Comment by zi0Black** [comment]
I sent you the rootfs on TG.

**Comment by wtdcode** [comment]
Still stuck in this issue and temporarily mark it as stalled.

**Comment by wtdcode** [comment]
Progress: Confirmed that it is related to arm `stat64` implementation.

**Comment by wtdcode** [comment]
Okay,  the wrong implementation of `stat64` results in `ld.so` not loading the object correctly.

**Comment by wtdcode** [comment]
Fixed in 6a30e9e4c1eebe4d3338fab47af875ea7d0f6bcb
